lib/galaxy/tool_util/models.py

```diff
@@ -64,6 +72,28 @@ class UserToolSource(BaseModel):
     xrefs: Optional[List[XrefDict]] = None
     help: Optional[HelpContent] = None
 
+    @model_validator(mode="before")
+    @classmethod
+    def normalize_items(cls, values):
+        if isinstance(values, dict):
+            normalize_dict(values, ["inputs", "outputs"])
+        return values
+
+
+class UserToolSource(ToolSourceBase):
+    class_: Annotated[Literal["GalaxyUserTool"], Field(alias="class")]
+    name: str
+    shell_command: str
+    container: str
+
+
+class AdminToolSource(ToolSourceBase):
```

Comment:
Totally fine just dropping the Admin version here if it is easier. The point was to establish some abstractions that could be used by the feature you're implementing and CWL - you've gone a lot farther with the user tools so I'm happy to drop any old admin-only, insecure functionality if it would help. Either way though is fine.